io/gdal/GDALWriter.cpp

```diff
@@ -0,0 +1,187 @@
+        "http://pdal.io/stages/writers.gdal.html");
+
+
+CREATE_STATIC_PLUGIN(1, 0, GDALWriter, Writer, s_info);
+
+std::string GDALWriter::getName() const
+{
+    return s_info.name;
+}
+
+
+void GDALWriter::addArgs(ProgramArgs& args)
+{
+    args.add("filename", "Output filename", m_filename).setPositional();
+    args.add("edge_length", "Length of cell edges (cells are square)",
+        m_edgeLength).setPositional();
+    args.add("radius", "Radius from cell center to use to locate influencing "
+        "points", m_radius).setPositional();
+    args.add("gdaldriver", "GDAL writer driver name", m_drivername, "GTiff");
+    args.add("gdalopts", "GDAL driver options (name=value,name=value...)",
+        m_options);
+    args.add("output_type", "Statistics produced ('min', 'max', 'mean', "
+        "'idw', 'count', 'stdev' or 'all')", m_outputTypeString);
+    args.add("window_size", "Cell distance for fallback interpolation",
+        m_windowSize);
+}
+
+
+void GDALWriter::initialize()
+{
+    for (auto& ts : m_outputTypeString)
+    {
+       Utils::trim(ts);
+        if (ts == "all")
+        {
+            m_outputTypes = ~0;
+            break;
+        }
+        if (ts == "min")
+            m_outputTypes |= GDALGrid::statMin;
+        else if (ts == "max")
+            m_outputTypes |= GDALGrid::statMax;
+        else if (ts == "count")
+            m_outputTypes |= GDALGrid::statCount;
+        else if (ts == "mean")
+            m_outputTypes |= GDALGrid::statMean;
+        else if (ts == "idw")
+            m_outputTypes |= GDALGrid::statIdw;
+        else if (ts == "stdev")
+            m_outputTypes |= GDALGrid::statStdDev;
+        else
+        {
+            std::ostringstream oss;
+            oss << "Invalid writers.gdal output type: '" << ts << "'.";
+            throw pdal_error(oss.str());
+        }
+    }
+    if (m_outputTypes == 0)
+        m_outputTypes = ~0;
+
+    gdal::registerDrivers();
+}
+
+void GDALWriter::ready(PointTableRef table)
+{
+    if (!table.spatialReferenceUnique())
+    {
+        std::ostringstream oss;
+
+        oss << getName() << ": Can't write output with multiple spatial "
+            "references.";
+        throw pdal_error(oss.str());
+    }
+}
+
+
+void GDALWriter::write(const PointViewPtr view)
+{
+    view->calculateBounds(m_bounds);
+    size_t width = ((m_bounds.maxx - m_bounds.minx) / m_edgeLength) + 1;
+    size_t height = ((m_bounds.maxy - m_bounds.miny) / m_edgeLength) + 1;
+    m_grid.reset(new GDALGrid(width, height, m_edgeLength, m_radius, -9999.0,
+        m_outputTypes, m_windowSize));
+
+    for (PointId idx = 0; idx < view->size(); ++idx)
+    {
+        double x = view->getFieldAs<double>(Dimension::Id::X, idx) -
+            m_bounds.minx;
+        double y = view->getFieldAs<double>(Dimension::Id::Y, idx) -
+            m_bounds.miny;
+        double z = view->getFieldAs<double>(Dimension::Id::Z, idx);
+
+        m_grid->addPoint(x, y, z);
```

Comment:
As described above, it would be really handy to be able to specify any dimension here, not just Z.